src/sentry/static/sentry/app/components/charts/optionSelector.tsx

```diff
@@ -19,48 +19,82 @@ type Props = {
   menuWidth?: string;
 };
 
-function OptionSelector({options, onChange, selected, title, menuWidth = 'auto'}: Props) {
-  const selectedOption = options.find(opt => selected === opt.value) || options[0];
-
-  return (
-    <InlineContainer>
-      <SectionHeading>{title}</SectionHeading>
-      <MenuContainer>
-        <DropdownMenu alwaysRenderMenu={false}>
-          {({isOpen, getMenuProps, getActorProps}) => (
-            <React.Fragment>
-              <StyledDropdownButton {...getActorProps()} size="zero" isOpen={isOpen}>
-                {selectedOption.label}
-              </StyledDropdownButton>
-              <StyledDropdownBubble
-                {...getMenuProps()}
-                alignMenu="right"
-                width={menuWidth}
-                isOpen={isOpen}
-                blendWithActor={false}
-                blendCorner
-              >
-                {options.map(opt => (
-                  <DropdownItem
-                    key={opt.value}
-                    onSelect={onChange}
-                    eventKey={opt.value}
-                    disabled={opt.disabled}
-                    isActive={selected === opt.value}
-                    data-test-id={`option-${opt.value}`}
-                  >
-                    <Tooltip title={opt.tooltip} containerDisplayMode="inline">
-                      {opt.label}
-                    </Tooltip>
-                  </DropdownItem>
-                ))}
-              </StyledDropdownBubble>
-            </React.Fragment>
-          )}
-        </DropdownMenu>
-      </MenuContainer>
-    </InlineContainer>
-  );
+type State = {
+  menuContainerWidth?: number;
+};
+
+class OptionSelector extends React.Component<Props, State> {
+  state: State = {};
+
+  componentDidMount() {
+    this.setMenuContainerWidth();
+  }
+
+  shouldComponentUpdate(nextProps: Props, nextState: State) {
+    return !isEqual(nextProps, this.props) || !isEqual(nextState, this.state);
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    if (prevProps.selected !== this.props.selected) {
+      this.setMenuContainerWidth();
+    }
+  }
+
+  setMenuContainerWidth() {
+    const menuContainerWidth = this.menuContainerRef?.current?.offsetWidth;
+    if (menuContainerWidth) {
+      this.setState({menuContainerWidth});
+    }
+  }
+
+  menuContainerRef = React.createRef<HTMLDivElement>();
+
+  render() {
+    const {menuContainerWidth} = this.state;
+    const {options, onChange, selected, title, menuWidth = 'auto'} = this.props;
```

Comment:
Should this be in default props instead?

Reply:
sure thing! let me update it.